src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsCredentialProbeManager.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ImdsCredentialProbeManager
+    {
+        private const string CredentialEndpoint = "http://169.254.169.254/metadata/identity/credential";
+        private const string ProbeBody = ".";
+        private const string ImdsHeader = "IMDS/";
+        private readonly IHttpManager _httpManager;
+        private readonly ILoggerAdapter _logger;
+
+        public ImdsCredentialProbeManager(IHttpManager httpManager, ILoggerAdapter logger)
+        {
+            _httpManager = httpManager ?? throw new ArgumentNullException(nameof(httpManager));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<bool> ExecuteAsync(CancellationToken cancellationToken = default)
+        {
+            _logger.Info("[Credential Probe] Initiating probe to IMDS credential endpoint.");
+
+            var request = new ManagedIdentityRequest(HttpMethod.Post, new Uri($"{CredentialEndpoint}?cred-api-version=1.0"))
+            {
+                Content = ProbeBody
+            };
+
+            HttpContent httpContent = request.CreateHttpContent();
+
+            _logger.Info($"[Credential Probe] Sending request to {CredentialEndpoint}");
+            _logger.Verbose(() => $"[Credential Probe] Request Headers: {string.Join(", ", request.Headers)}");
```

Comment:
It's a little dangerous to log headers and body. Maybe don't do it unless we really need it.